File: app/src/main/java/net/wojteksz128/worktimemeasureapp/notification/endOfWork/EndOfWorkAction.java
package net.wojteksz128.worktimemeasureapp.notification.endOfWork;

import android.content.Context;
import android.widget.Toast;

import net.wojteksz128.worktimemeasureapp.notification.NotificationAction;

class EndOfWorkAction implements NotificationAction {

    @Override
    public void invoke(Context context) {
        // TODO: 03.11.2018 implement this
        Toast.makeText(context, "EndOfWorkAction", Toast.LENGTH_LONG).show();
    }
}

File: app/src/main/kotlin/net/wojteksz128/worktimemeasureapp/notification/endOfWork/EndOfWorkAction.kt
package net.wojteksz128.worktimemeasureapp.notification.endOfWork

import android.content.Context
import android.os.Handler
import android.os.Looper
import android.widget.Toast

import net.wojteksz128.worktimemeasureapp.R
import net.wojteksz128.worktimemeasureapp.util.notification.NotificationAction
import net.wojteksz128.worktimemeasureapp.util.notification.NotificationUtils

internal class EndOfWorkAction : NotificationAction {

    override fun invoke(context: Context) {
        // TODO: 03.11.2018 implement this
        NotificationUtils.clearAllNotifications(context)

        val handler = Handler(Looper.getMainLooper())
        handler.post { Toast.makeText(context.applicationContext, context.getString(R.string.dashboard_snackbar_info_outcome_registered), Toast.LENGTH_LONG).show() }
    }
}
